File: memory-domain/src/main/java/com/memory/dto/memberlink/response/MemberPublicLinkResponse.java
package com.memory.dto.memberlink.response;

import com.memory.dto.member.response.MemberResponse;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.List;

@Getter
@AllArgsConstructor
public class MemberPublicLinkResponse {

    private List<MemberLinkResponse> memberLinks;
    private MemberResponse member;

    public static MemberPublicLinkResponse of(List<MemberLinkResponse> memberLinks, MemberResponse member) {
        return new MemberPublicLinkResponse(memberLinks, member);
    }

}

File: memory-api/src/main/kotlin/com/memory/dto/memberlink/response/MemberPublicLinkResponse.kt
package com.memory.dto.memberlink.response

import com.memory.dto.member.response.MemberResponse

data class MemberPublicLinkResponse(
    val memberLinks: MutableList<MemberLinkResponse?>?,
    val member: MemberResponse?
) {
    companion object {
        @JvmStatic
        fun of(memberLinks: MutableList<MemberLinkResponse?>?, member: MemberResponse?): MemberPublicLinkResponse {
            return MemberPublicLinkResponse(memberLinks, member)
        }
    }
}
